### PR TITLE
Generate unique node names

### PR DESCRIPTION
**Pull request by ThomasMatern**
Append the port to the node name, e.g. /serial_node_dev_ttyACM0 to allow running multiple instances on the same host.

### CODE REVIEW

File: rosserial_python/nodes/serial_node.py [hunk 1]
```diff
@@ -59,6 +56,9 @@
         port_name  = sys.argv[1]
     if len(sys.argv) == 3 :
         tcp_portnum = int(sys.argv[2])
+
+    rospy.init_node("serial_node{}".format(port_name.replace('/', '_')))
+    rospy.loginfo("ROS Serial Python Node")
```
Comment (mikepurvis) [inline rosserial_python/nodes/serial_node.py:61]: Are you able to access private params prior to calling `rospy.init_node`?

**Comment by mikepurvis** [comment]
What is gained here over using `name="$(anon serial_node)"` in the roslaunch file?

**Comment by ThomasMatern** [comment]
Sorry, I'm relatively new to ROS, just trying to find my way around.
I am running the serial_node directly from udev and pass the device name as a parameter, using rosrun directly, so there is no roslaunch file.

**Comment by mikepurvis** [comment]
Cool, welcome!

Running a ros process directly from udev is certainly possible, but not typical or recommended. Some issues you're likely to encounter:
- Processes running from udev run as root, which you probably don't want for ROS.
- When something goes wrong, you'll be pretty blind— I don't believe udev captures stdout anywhere for processes it spawns.

The udev RUN capability is more about discovering names and such for devices; it's not meant to launch long-running processes.

**Pull request closed without merging**